src/aurelia-event-aggregator.ts

```diff
@@ -0,0 +1,168 @@
+import * as LogManager from 'aurelia-logging';
+
+const logger = LogManager.getLogger('event-aggregator');
+
+export type Constructor<T = object> = new (...args: any[]) => T;
+export type EventCallback<T = any> = (data: T, event?: string) => any;
+
+export class Handler {
+  constructor(public messageType: Constructor, public callback: EventCallback) {}
+
+  public handle(message: object) {
+    if (message instanceof this.messageType) {
+      this.callback.call(null, message);
+    }
+  }
+}
+
+function invokeCallback(callback: EventCallback, data: any, event: string) {
+  try {
+    callback(data, event);
+  } catch (e) {
+    logger.error(e);
+  }
+}
+
+function invokeHandler(handler: Handler, data: object) {
+  try {
+    handler.handle(data);
+  } catch (e) {
+    logger.error(e);
+  }
+}
+
+/**
+ * Represents a disposable subsciption to an EventAggregator event.
+ */
+export interface Subscription {
+  /**
+   * Disposes the subscription.
+   */
+  dispose(): void;
+}
+
+/**
+ * Enables loosely coupled publish/subscribe messaging.
+ */
+export class EventAggregator {
+  /**
+   * Creates an instance of the EventAggregator class.
+   */
+  constructor(
+    private eventLookup: {[event: string]: EventCallback[]} = {},
```

Comment:
What is the purpose of the constructor API change? You also had to `export` the `Handler` class because of this.

Reply:
I was curious about this as well. @mjwwit Can you provide some insight? Thanks!

Reply:
You guys are absolutely correct, I was fiddling around with this during the migration and accidentally left it in. I'll fix this asap.